### PR TITLE
ca-email-confirm web component

### PR DESCRIPTION
**Pull request by kevmarchant**

### CODE REVIEW

File: components/ca-email-confirm.js [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+            this.render({ [name.toString()]: newValue });
+        }
+
+        /**
+         * returns the entered email address, if they both match and are valid email addresses
+         * @return {string} the entered email address if valid
+         */
+        get value() {
+
+            if (!this.validate()) return '';
```
Comment (kevmarchant) [inline components/ca-email-confirm.js]: yup!

File: components/helpers/types.js [hunk 2]
```diff
@@ -8,7 +9,18 @@ define('types', [], () => {
         return value === 'true';
     }
 
+    /**
+     * isValidEmailAddress - returns if email passes regex validation
+     * @param {string} emailAddress - email address string to validate
+     * @return {boolean} email address is valid
+     */
+    function isValidEmailAddress(emailAddress) {
+        const regex = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
```
Comment (kevmarchant) [inline components/helpers/types.js]: it can be swapped out easily.  is there a better one you're aware of?

**Comment by norisalsaadie** [comment]
+1, clean. I am not a fan of padded-blocks but the linting rules have disabled any feedback on this.

File: components/helpers/types.js (hunk 2 above)
Comment (norisalsaadie) [inline components/helpers/types.js]: It's fine :)

**Comment by joeburton** [comment]
@kevmarchant there are conflicts in the branch that need fixing

File: components/ca-email-confirm.js [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+            this.render({ [name.toString()]: newValue });
+        }
+
+        /**
+         * returns the entered email address, if they both match and are valid email addresses
+         * @return {string} the entered email address if valid
+         */
+        get value() {
+
+<<<<<<< c8ab26050ed67be07f464ce9f07710325eeed04f
```
Comment (norisalsaadie) [inline components/ca-email-confirm.js]: conflicts

**Pull request merged**